src/cpp/main.cpp

```diff
@@ -47,32 +51,48 @@ size_t initialize(const Config *config) {
   displacements = (Vector*) calloc(num_particles, sizeof(Vector));
 
   // Initialize the particles.
+
+  // Particles radious generator in milimeters
+  int seed = 7;
+  std::mt19937 generator(seed);
+  std::uniform_real_distribution<double> uniform_distribution(min_radious, max_radious);
+
+  // In README.md says that m is the parameter for the Y axis ??
   const unsigned int max_in_x = floor(config->m / diameter);
-  double x = config->r;
-  double y = config->r;
+  double current_radious = uniform_distribution(generator);
+  double last_radious;
+  double particles_last_radious[max_in_x] = {0}; // Radious of particle below the particle currently positioned
+  double x = current_radious;
+  double y = current_radious;
   for (size_t i = 1; i < num_particles; ++i) {
+
     particles[i].x_coordinate = x;
     particles[i].y_coordinate = y;
-    particles[i].radious = config->r;
+    particles[i].radious = current_radious;
     properties[i].mass = config->mass;
     properties[i].kn = config->kn;
     properties[i].ks = config->ks;
 
+    // Variable update
+    y += particles_last_radious[i % max_in_x];
+    particles_last_radious[i % max_in_x] = current_radious;
+    last_radious = current_radious;
+    current_radious = uniform_distribution(generator);
+
     // Check if this particle is the last one for this row...
     if ((i % max_in_x) == 0) {
-      // If it is, reset the x value and increment y.
-      x = config->r;
-      y += diameter;
+      // If it is, reset the x value
+      x = current_radious;
     } else {
-      // If not, increment the x value.
-      x += diameter;
+      // Else, increment the x value.
+      x += (last_radious + current_radious);
     }
   }
 
   // Initialize the falling particle.
+  particles[0].radious = uniform_distribution(generator);
   particles[0].x_coordinate = config->m / 2;
-  particles[0].y_coordinate = config->n + (4 * config->r);
-  particles[0].radious = config->r;
+  particles[0].y_coordinate = config->n + (4 * particles[0].radious);
```

Comment:
I believe this could be changed by a constant. Like `n + 80` or  something like that.
Also, I am not sure if the radius of the falling particle should be random too, we should discuss that with the team.

Finally, I just noticed it should be **radius** instead of **radious** 😲 
Can you help us with that change?

Reply:
No problem, I'll change that.